src/EditorFeatures/Core/Implementation/AutomaticCompletion/AbstractAutomaticLineEnderCommandHandler.cs

```diff
@@ -84,54 +80,53 @@ public void ExecuteCommand(AutomaticLineEnderCommandArgs args, Action nextHandle
                 return;
             }
 
-            _waitIndicator.Wait(
-                title: EditorFeaturesResources.Automatic_Line_Ender,
-                message: EditorFeaturesResources.Automatically_completing,
-                allowCancel: false, action: w =>
+            using (context.WaitContext.AddScope(allowCancellation: false, EditorFeaturesResources.Automatically_completing))
+            {
+                var userCancellationToken = context.WaitContext.UserCancellationToken;
+
+                // caret is not on the subject buffer. nothing we can do
+                var position = args.TextView.GetCaretPoint(args.SubjectBuffer);
+                if (!position.HasValue)
+                {
+                    NextAction(operations, nextHandler);
+                    return;
+                }
+
+                var subjectLineWhereCaretIsOn = position.Value.GetContainingLine();
+                var insertionPoint = GetInsertionPoint(document, subjectLineWhereCaretIsOn, userCancellationToken);
+                if (!insertionPoint.HasValue)
+                {
+                    NextAction(operations, nextHandler);
+                    return;
+                }
+
+                // special cases where we treat this command simply as Return.
+                if (TreatAsReturn(document, position.Value.Position, userCancellationToken))
                 {
-                    // caret is not on the subject buffer. nothing we can do
-                    var position = args.TextView.GetCaretPoint(args.SubjectBuffer);
-                    if (!position.HasValue)
-                    {
-                        NextAction(operations, nextHandler);
-                        return;
-                    }
-
-                    var subjectLineWhereCaretIsOn = position.Value.GetContainingLine();
-                    var insertionPoint = GetInsertionPoint(document, subjectLineWhereCaretIsOn, w.CancellationToken);
```

Comment:
>w.CancellationToken [](start = 96, length = 19)

ugh, another case of disabled cancellation, but used token. I'll replacer it with CancellationToken.None for clarity